### ISSUE TITLE
bug: globalStyle failing HMR on Windows

### ISSUE THREAD

**Issue by 4leite** [ionic-team/stencil#4961]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.5.0

### Current Behavior

Stencil in watch mode does not rebuild when config.globalStyle is updated on windows.

### Expected Behavior

Changing config.globalStyle should trigger a rebuild.

### System Info

```shell
Latest Windows (this is relevant because the bug will not be present on linux).
```

### Steps to Reproduce

npm run start
make a change to /src/global.css eg change "--color-primary: red;" to blue.
change is not reflected in browser.

### Code Reproduction URL

https://github.com/Datacom-Digital/stencil-bug

### Additional Information

 This is because buildCtx.filesChanged is normalised but config.globalStyle is not so the string compare fails. This is probably effecting imports as well however i don't use them in my project so haven't checked.
 
 See #4898  for the patch that fixed the issue for me.

**Comment by 4leite** [comment]
@rwaskiewicz  this effects both watch and regular build (it's not an HMR issue), it's a file diff problem.

I am happy to raise a PR with the one line fix I linked above that resolves the issue.

**Comment by rwaskiewicz** [comment]
(Still working on this)

**Comment by rwaskiewicz** [comment]
@4leite  Thanks for your patience!

Can you do me a favor and give the following dev build of Stencil a try when you get a chance?
```
npm i @stencil/core@4.7.1-dev.1699535671.a4c3950
```

**Comment by tanner-reits** [comment]
The fix for this went out as a part of today's [v4.7.2 release](https://github.com/ionic-team/stencil/releases/tag/v4.7.2)!

**Comment by 4leite** [comment]
@tanner-reits  Sorry took a while to get back on this project. I can confirm that this is resolved for ours with release v4.7.2.
Thanks.
